HousingSearchApi.Tests/V1/E2ETests/Steps/GetAssetSteps.cs

```diff
@@ -247,6 +254,13 @@ public async Task ThenAssetsWithProvidedContractStatusShouldBeIncluded(string co
             result.Results.Assets.Count().Should().Be(expectedNumberOfAssets);
             result.Results.Assets.All(x => x.AssetContract.IsActive == bool.Parse(contractStatus));
         }
+        public async Task ThenAssetsWhoseContractHasProvidedChargesSubytpeAreReturned(string chargesSubtype, int expectedNumberOfAssets)
+        {
+            var resultBody = await _lastResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+            var result = JsonSerializer.Deserialize<APIAllResponse<GetAllAssetListResponse>>(resultBody, _jsonOptions);
+
+            result.Results.Assets.Count().Should().Be(expectedNumberOfAssets);
+        }
```

Comment:
Unlike the name of this method suggests, this step does check whether the **correct** assets are returned.

This only checks for count, so in theory, if the implementation was broken & it returned something else entirely, so long as there's exactly item returned, this test would pass. :thinking:

Reply:
Addressed in [4fc36b0](https://github.com/LBHackney-IT/housing-search-api/pull/235/commits/4fc36b0e099876ac21ca51d32ac1fb656cb2726b).